lib/Service/LoginService.php

```diff
@@ -253,6 +260,22 @@ public function login($profile, $userSession, $request)
                 }
             }
 
+            if ($profile[$attr['photoURL']]) {
+                try {
+                    $curl = curl_init($profile[$attr['photoURL']]);
+                    curl_setopt($curl, CURLOPT_HEADER, false);
+                    curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+                    curl_setopt($curl, CURLOPT_BINARYTRANSFER, true);
+                    curl_setopt($curl, CURLOPT_USERAGENT, 'Mozilla/5.0');
+                    $raw = curl_exec($curl);
+                    curl_close($curl);
```

Comment:
I'm not a huge fan of the picture claim, as it basically forces the RP to request a possibly user supplied URL. We need to be extra cautious here to not introduce a SSRF vulnerability similar to [CVE-2021-26715](https://nvd.nist.gov/vuln/detail/CVE-2021-26715) for example.